lib/rubocop/cop/chef/effortless/chef_vault_used.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+#
+# Copyright:: 2019, Chef Software Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module RuboCop
+  module Cop
+    module Chef
+      module ChefEffortless
+        # Cookbook:: Chef Vault does not work with Effortless
+        #
+        # @example
+        #
+        #   # bad
+        #   require 'chef-vault'
+        #
+        #   # bad
+        #   ChefVault::Item
+        #
+        #   # bad
+        #   include_recipe 'chef-vault'
+        #
+        #   # bad
+        #   chef_gem 'chef-vault'
+        #
+        #   # bad
+        #   chef_vault_item_for_environment(arg, arg1)
+        #
+        #   # bad
+        #   chef_vault_item(arg, arg1)
+        #
+        class ChefVaultUsed < Base
+          MSG = 'Chef Vault usage is not supported in the Effortless pattern'
+          RESTRICT_ON_SEND = [:chef_vault_item,
+                              :chef_vault_item_for_environment,
+                              :include_recipe,
+                              :require,
+                              :chef_gem].freeze
+
+          def_node_matcher :require?, <<-PATTERN
+            (send nil? :require
+              (str "chef-vault"))
+          PATTERN
+
+          def_node_matcher :include?, <<-PATTERN
```

Comment:
Since the string is the same on these you could turn them into a single matcher { :require :include_recipe :chef_gem}. It's probably a bit faster to process a single matcher.

Reply:
Awesome! I didn't realize I could do that. Updated!